apps/asap-server/src/handlers/webhooks/webhook-events-updated.ts

```diff
@@ -1,11 +1,105 @@
 
-const logger = debug('asap-server');
+import { Handler } from '../../utils/types';
+import { http } from '../../utils/instrumented-framework';
+import Calendars, { CalendarController } from '../../controllers/calendars';
+import {
+  syncCalendarFactory,
+  SyncCalendarFactory,
+} from '../../utils/sync-google-calendar';
+import getJWTCredentials, {
+  GetJWTCredentials,
+} from '../../utils/aws-secret-manager';
+import logger from '../../utils/logger';
 
-export const handler: APIGatewayProxyHandlerV2<void> = async (
-  event,
-  context,
-) => {
-  logger('Request received', JSON.stringify({ event, context }, null, 2));
+export const webhookEventUpdatedHandlerFactory = (
+  calendars: CalendarController,
+  getJWTCredentials: GetJWTCredentials,
+  syncCalendarFactory: SyncCalendarFactory,
+): Handler =>
+  http(
+    async (request: lambda.Request): Promise<lambda.Response> => {
+      // TODO: validate google request auth
+
+      const resourceId = request.headers['x-goog-resource-id'];
+      if (!resourceId) {
+        return {
+          statusCode: 400,
+          payload: {
+            message: 'Missing x-goog-resource-id header',
+          },
+        };
+      }
+
+      const calendar = await calendars
+        .fetchByResouceId(resourceId)
+        .catch((err) => {
+          logger('Error fetching calendar', err);
+          return undefined;
+        });
+
+      if (!calendar?.id) {
+        return { statusCode: 502 };
+      }
+      const squidexCalendarId = calendar.id;
+      const googleCalendarId = calendar.data.id.iv;
+
+      const syncToken = await calendars
+        .getSyncToken(squidexCalendarId)
+        .catch((err) => {
+          logger('Error fetching syncToken', err);
+          return undefined;
+        });
+
+      const credentials = await getJWTCredentials().catch((err) => {
+        logger('Error fetching AWS credentials', err);
+        return undefined;
+      });
+
+      if (!credentials) {
+        return { statusCode: 502 };
+      }
+
+      const auth = new Auth.GoogleAuth({
+        scopes: [
+          'https://www.googleapis.com/auth/calendar',
+          'https://www.googleapis.com/auth/calendar.events',
+        ],
+      }).fromJSON(credentials) as Auth.JWT;
+
+      const syncCalendar = syncCalendarFactory(
+        syncToken,
+        syncEventFactory(squidexCalendarId),
+        auth,
+      );
+      const nextSyncToken = await syncCalendar(googleCalendarId);
+
+      if (nextSyncToken) {
+        await calendars
+          .update(squidexCalendarId, { syncToken: nextSyncToken })
+          .catch((err) => {
+            logger('Error updated syncToken', err);
+          });
+      }
+
+      return {
+        statusCode: 200,
+      };
+    },
+  );
+
+export const handler: Handler = webhookEventUpdatedHandlerFactory(
+  new Calendars(),
+  getJWTCredentials,
+  syncCalendarFactory,
+);
+
+const syncEventFactory = (calendarId: string) => {
```

Comment:
since you moved syncCalendar to a separate file, do you think maybe this should be moved with it? just a suggestion, your call

Reply:
That will happen in my next PR 😉  I just wanted to make it simple on this one.